core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -169,6 +171,7 @@ private[spark] class TaskSchedulerImpl(
   protected val executorIdToHost = new HashMap[String, String]
 
   private val abortTimer = ThreadUtils.newDaemonSingleThreadScheduledExecutor("task-abort-timer")
+  private val abortFutures = new CopyOnWriteArrayList[ScheduledFuture[_]]
```

Comment:
Why CopyOnWriteArrayList in particular? maybe fine but would a simple synchronized list be easier?

Reply:
Yeah. less reads and more writes here.